### PR TITLE
fetch all environments unpaginated

### PR DESCRIPTION
**Pull request by Suyash-d**
**Description**
create a new GET /environment/list endpoint allowing to the environments to be fetched in a non paginated way. This is needed to allow the UI to fetch ALL of the environments when the user wants to execute a script.

extend api doc from metadew/iesi-api-documentation#22
extend code from #352

### CODE REVIEW

**Comment by Suyash-d** [comment]
This PR to be merged in this branch before merged to develop ->  https://github.com/metadew/iesi/pull/368

**Comment by BerrevoetsRobbe** [comment]
to be merged together with metadew/iesi-ui#101

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/configuration/security/jwt/EnvironmentsControllerSecurityTest.java [hunk 1]
```diff
@@ -65,7 +65,7 @@
 
     @Test
     void testGetAllNoUser() throws Exception {
-        assertThatThrownBy(() -> environmentsController.getAll(Pageable.unpaged()))
+        assertThatThrownBy(() -> environmentsController.getAll())
                 .isInstanceOf(AuthenticationCredentialsNotFoundException.class);
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/configuration/security/jwt/EnvironmentsControllerSecurityTest.java:69]: don't change an existing test when adding a new method. Write a new test instead

File: core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/configuration/security/jwt/EnvironmentsControllerSecurityTest.java [hunk 2]
```diff
@@ -105,8 +105,7 @@ void testGetAllNoEnvironmentReadPrivilege() throws Exception {
     @WithIesiUser(username = "spring",
             authorities = {"ENVIRONMENTS_READ@PUBLIC"})
     void testGetEnvironmentReadPrivilege() throws Exception {
-        Pageable pageable = Pageable.unpaged();
-        environmentsController.getAll(pageable);
+        environmentsController.getAll();
     }
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/configuration/security/jwt/EnvironmentsControllerSecurityTest.java:109]: don't change an existing test when adding a new method. Write a new test instead

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/environment/EnvironmentsController.java [hunk 3]
```diff
@@ -66,6 +65,16 @@
...
 
+    @GetMapping("/list")
+    @PreAuthorize("hasPrivilege('ENVIRONMENTS_READ')")
+    public HalMultipleEmbeddedResource<EnvironmentDto> getAll() {
+        List<Environment> environments = environmentService.getAll();
+        return new HalMultipleEmbeddedResource<>(
+                environments.stream().filter(distinctByKey(Environment::getName))
+                        .map(environment -> environmentDtoResourceAssembler.toModel(environment))
+                        .collect(Collectors.toList()));
+    }
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/environment/EnvironmentsController.java:76]: can't you use environmentDtoService#getAll(Page.unpaged) instead?

Comment (Suyash-d) [reply core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/environment/EnvironmentsController.java]: Done

File: core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/environment/EnvironmentsController.java [hunk 4]
```diff
@@ -66,6 +65,20 @@
         return (PagedModel<EnvironmentDto>) environmentDtoPagedResourcesAssembler.toEmptyModel(environmentDtoPage, EnvironmentDto.class);
     }
 
+    @GetMapping("/list")
+    @PreAuthorize("hasPrivilege('ENVIRONMENTS_READ')")
+    public HalMultipleEmbeddedResource<EnvironmentDto> getAll() {
+        List<EnvironmentDto> environmentDtos = environmentDtoService.getAll();
```
Comment (BerrevoetsRobbe) [inline core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/environment/EnvironmentsController.java:71]: can't you call the public Page<EnvironmentDto> getAll(Pageable pageable) and provide the Pageable.unpaged() as an argument?

**Pull request merged**